underpants.go

```diff
@@ -472,8 +472,19 @@ func setup(c *conf, port int) (*http.ServeMux, error) {
       panic(err)
     }
 
+    uSplitEmail := strings.SplitN(u.Email, "@", 2)
+    if len(uSplitEmail) != 2 {
+      http.Error(w,
+        http.StatusText(http.StatusForbidden),
+        http.StatusForbidden)
+      return
+    }
+
+    uDomain := uSplitEmail[1]
+
     // this only happens when someone edits the auth url
-    if !strings.HasSuffix(u.Email, c.Oauth.Domain) {
+    if uDomain != c.Oauth.Domain &&
+        !strings.HasSuffix(uDomain, "." + c.Oauth.Domain) {
```

Comment:
I thought that was why the check was written like this in the first place, I'll change it